spark/src/main/scala/geotrellis/spark/pyramid/Pyramid.scala

```diff
@@ -42,6 +65,69 @@ object Pyramid extends LazyLogging {
     implicit def methodToOptions(m: ResampleMethod): Options = Options(resampleMethod = m)
   }
 
+  def read[
+    K: AvroRecordCodec: Boundable: JsonFormat: ClassTag: SpatialComponent,
+    V <: CellGrid: ? => TilePrototypeMethods[V]: ? => TileMergeMethods[V]: AvroRecordCodec: ClassTag,
+    M: JsonFormat: Component[?, Bounds[K]]: Component[?, LayoutDefinition]
+  ](layerName: String, maxZoom: Int, minZoom: Int, layerReader: LayerReader[LayerId]): Pyramid[K, V, M] = {
+    val seq = for (z <- maxZoom to minZoom by -1) yield {
+      (z, layerReader.read[K, V, M](LayerId(layerName, z)))
+    }
+    Pyramid[K, V, M](seq.toMap)
+  }
+
+  def write[
```

Comment:
Tempting to put this on `FilteringLayerReader` trait if we're adding this type as first class citizen.